client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -244,74 +206,6 @@ Undocumented.prototype.getDomainPrice = function ( domain, fn ) {
 	);
 };
 
-function mapKeysRecursively( object, fn ) {
-	return Object.keys( object ).reduce( function ( mapped, key ) {
-		let value = object[ key ];
-		if ( isPlainObject( value ) ) {
-			value = mapKeysRecursively( value, fn );
-		}
-
-		mapped[ fn( key ) ] = value;
-		return mapped;
-	}, {} );
-}
-
-/**
- * Validates the specified domain contact information against a list of domain names.
- *
- * @param {object} contactInformation - user's contact information
- * @param {string[]} domainNames - list of domain names
- * @param {Function} fn The callback function
- * @param {object} query Query object for the call to wpcom.req.post
- */
-Undocumented.prototype.validateDomainContactInformation = function (
-	contactInformation,
-	domainNames,
-	fn,
-	query
-) {
-	let data = {
-		contactInformation: contactInformation,
-		domainNames: domainNames,
-	};
-
-	debug( '/me/domain-contact-information/validate query' );
-	data = mapKeysRecursively( data, snakeCase );
-
-	// Due to backend limitations some versions of this endpoint
-	// serialize a nested object in the response, encoding e.g.
-	//   { foo: { bar: { baz: [ "error" ] } } }
-	// as
-	//   { foo.bar.baz: [ "error" ] }
-	// here we decide whether to rehydrate this.
-	const shouldReshapeResponse = query?.apiVersion === '1.2';
```

Comment:
There's one suspicious thing about the `/me/domain-contact-information/validate` endpoint: it has version 1.2, but we're using the v1.1 one.

If I understand the PHP implementation correctly, the difference is the format of nested validation error messages in the response.

the new v1.2:
```
messages: {
  "extra.ca.cira_agreement_accepted": [
    "Please review and accept the CIRA Registrant Agreement."
  ]
}
```
in the old v1.1 only the last part of the name is sent, possibly ambiguous:
```
messages: {
  "cira_agreement_accepted": [
    "Please review and accept the CIRA Registrant Agreement."
  ]
}
```

@sirbrillig Do you know if the REST response really has the shape we expect? The `composite-checkout` package has some tests for these `extra` domain fields.

Reply:
We are still using the 1.2 endpoint FWIW: https://github.com/Automattic/wp-calypso/blob/b6894d955505e48f0113dc8e655aa3aba3ba749f/client/my-sites/checkout/composite-checkout/lib/contact-validation.tsx#L310

I'll proceed and ship this as the previous version of the code was using the v1.1 of the endpoint, but I'm happy to revisit if there's the need for that.

Reply:
> If I understand the PHP implementation correctly, the difference is the format of nested validation error messages in the response.

You're correct about that. When we remade checkout, we wanted a flat version of the error messages to make it easier to iterate over them and assign them to our new managed data structure, so we created v1.2 of the endpoint for that purpose.

These forms predate the checkout changes and still use the formStateController mechanism for errors, so (I think) they require the old version of the endpoint.